frontend/src/components/PropertyCard.vue

```diff
@@ -79,8 +79,8 @@
       >
         <q-badge
           color="grey-3"
-          :text-color="property['SALE PRICE'] < estimated_price ? 'positive' : 'negative'"
-          :label="'$' + property['SALE PRICE'].toLocaleString()"
+          :text-color="property ? (property.salePrice < estimated_price ? 'positive' : 'negative') : null"
+          :label="property ? ('$' + property.salePrice.toLocaleString()) : null"
```

Comment:
It would be better, instead of checking for property on every attribute, to check for the whole component and if it's not loaded, display a loading indicator instead. Or, even better: We could build a placeholder with the same layout using e.g. https://www.npmjs.com/package/vue-shimmer